omnigibson/prims/rigid_prim.py

```diff
@@ -811,20 +818,24 @@ def _load_state(self, state):
         super()._load_state(state=state)
 
         # Set velocities if not kinematic
-        self.set_linear_velocity(np.array(state["lin_vel"]))
-        self.set_angular_velocity(np.array(state["ang_vel"]))
+        self.set_linear_velocity(
+            state["lin_vel"] if isinstance(state["lin_vel"], th.Tensor) else th.tensor(state["lin_vel"])
+        )
+        self.set_angular_velocity(
+            state["ang_vel"] if isinstance(state["ang_vel"], th.Tensor) else th.tensor(state["ang_vel"])
+        )
 
     def serialize(self, state):
         # Run super first
         state_flat = super().serialize(state=state)
 
-        return np.concatenate(
+        return th.cat(
             [
                 state_flat,
                 state["lin_vel"],
                 state["ang_vel"],
             ]
-        ).astype(float)
+        ).float()
```

Comment:
I probably have already mentioned this but let's do a global search of all of these type conversion, e.g. `double()`, `float()`, `int()`, and make sure these are all necessary.